### PR TITLE
feat(dash): Add react version of filmstrip controls

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Break out slider changes
- [x] Cross-browser testing
- [x] Unit tests

Depends on https://github.com/box/box-content-preview/pull/1410.

### CODE REVIEW

File: src/lib/viewers/media/DashViewer.js [hunk 1]
```diff
@@ -846,10 +851,22 @@ class DashViewer extends VideoBaseViewer {
...
-        if (filmstrip && filmstrip.metadata && filmstrip.metadata.interval > 0) {
+        const filmstripInterval = filmstrip && filmstrip.metadata && filmstrip.metadata.interval;
+
+        if (filmstripInterval > 0) {
             const url = this.createContentUrlWithAuthParams(filmstrip.content.url_template);
+
+            this.filmstripInterval = filmstripInterval;
             this.filmstripStatus = this.getRepStatus(filmstrip);
-            this.mediaControls.initFilmstrip(url, this.filmstripStatus, this.aspect, filmstrip.metadata.interval);
+            this.filmstripUrl = url;
```
Comment (ConradJChan) [inline src/lib/viewers/media/DashViewer.js:861]: Should these properties be added to the top of the class and documented?

File: src/lib/viewers/controls/media/Filmstrip.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
+@import './styles';
+
+.bp-Filmstrip {
```
[the remark this answers is not in the document]
Comment (jstoffan) [reply src/lib/viewers/controls/media/Filmstrip.scss:3]: I'm using a box-shadow to get the same effect without having to account for it in the sizing math.

File: src/lib/viewers/controls/media/TimeControls.tsx [hunk 3]
```diff
@@ -20,23 +25,53 @@ export const percent = (value1: number, value2: number): number => {
...
+                    time={hoverTime}
+                />
+            )}
+
             <SliderControl
                 className="bp-TimeControls-slider"
                 max={durationValue}
                 min={0}
+                onBlur={noop}
+                onFocus={noop}
```
[the remark this answers is not in the document]
Comment (jstoffan) [reply src/lib/viewers/controls/media/TimeControls.tsx:71]: It's either that or disable eslint checks for keyboard controls, which filmstrip doesn't support today.

[the remark this answers is not in the document]
Comment (ConradJChan) [reply src/lib/viewers/controls/media/TimeControls.tsx:71]: Maybe add a comment to explain that or an eslint ignore?

**Review by ConradJChan — APPROVED** [review]

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**